File: TWEB-project-SpringBoot/src/main/java/server/postgres/competitions/CompetitionsService.java
package server.postgres.competitions;

import org.springframework.stereotype.Service;
import server.postgres.clubs.Clubs;

import java.util.List;

@Service
public class CompetitionsService {
    public final CompetitionsRepository competitionsRepository;

    public CompetitionsService(CompetitionsRepository competitionsRepository) {
        this.competitionsRepository = competitionsRepository;
    }

    /**
     * Retrieve a list of competitions filtered by country.
     *
     * @param filter The country name to filter competitions.
     * @return A list of Competitions that occur in the specified country.
     */
    public List<Competitions> getCompetitionsByCountry(String filter) {
        List<Competitions> competitionsList = competitionsRepository.findByCountry(filter);
        return competitionsList;
    }

    /**
     * Retrieve a list of all competitions.
     *
     * @return A list of all Competitions.
     */
    public List<Competitions> getAllCompetition() {
        List<Competitions> competitionsList = competitionsRepository.getAllCompetition();
        return competitionsList;
    }

    /**
     * Retrieve a competition by its unique ID.
     *
     * @param competitionId The unique ID of the competition.
     * @return The Competitions object corresponding to the specified competition ID.
     */
    public Competitions getCompetitionById(String competitionId) {
        System.out.println("competitionId--->"+ competitionId);
        Competitions competitions = competitionsRepository.getCompetitionById(competitionId);
        return competitions;
    }

    /**
     * Retrieve a list of countries that have competitions.
     *
     * @return A list of country names as Strings, including an 'All' option.
     */
    public List<String> getCountry() {
        List<String> countryList = competitionsRepository.getCountry();
        countryList.add("All");
        return countryList;
    }

    /**
     * Retrieve a list of competition IDs.
     *
     * @return A list of competition IDs as Strings, including an 'All' option.
     */
    public List<String> getCompetitionsId() {
        List<String> competitionIdList = competitionsRepository.getCompetitionsId();
        competitionIdList.add("All");
        return competitionIdList;
    }

}

File: twebReact/src/services/competitionsService.js
import axios from 'axios';
const BASE_URL = 'http://localhost:3001/competitions';
/**
 * Fetches details for a competition by its ID.
 *
 * @param {number} competitionId - The unique identifier of the competition to retrieve details for.
 * @returns {Promise<object>} A promise that resolves with an object containing competition.
 */
export const getCompetitionsById = (competitionId) => {
  return axios.get(
    `${BASE_URL}/get-competitions-by-id?competitionId=${competitionId}`
  );
};
/**
 * Fetches details for all countries with competitions.
 *
 * @returns {Promise<Array>} A promise that resolves with an Array containing countries of competitions.
 */
export const getAllCountries = () => {
  return axios.get(`${BASE_URL}/get-competitions-country`);
};
/**
 * Fetches details for all competitions based on a filter.
 *
 * @param {string} filter - The filter for competitions.
 * @returns {Promise<Array>} A promise that resolves with an Array of competitions using the country as filter.
 */
export const getAllCompetitions = (filter) => {
  return axios.get(`${BASE_URL}/all-competitions?filter=${filter}`);
};
